### PR TITLE
[WIP] Parallelize cross-build jobs.

### PR DESCRIPTION
**Pull request by timoreimann**
This PR extends our cross-building script to support parallel execution.

The number of parallel jobs can be injected through the environment variable `MAX_PARALLEL_JOBS`. It defaults to 1 and is set to `TBD` when running in Travis.

### CODE REVIEW

**Comment by emilevauge** [comment]
Hey @timoreimann, could you provide a small description in your PR? Also, could you squash your commits?

File: script/crossbinary [hunk 1]
```diff
@@ -1,6 +1,29 @@
-#!/bin/bash
+#!/usr/bin/env bash
```
Comment (emilevauge) [inline script/crossbinary]: Could you explain why you need this ?

Comment (vdemeester) [inline script/crossbinary]: NixOS for example \o/ (where `/bin/bash` doesn' exists) but `/usr/bin/env` exists (this is almost the *one thing* that is almost always at the same place).

Comment (timoreimann) [inline script/crossbinary]: Yep, for better portability basically (see also [this SO answer](http://stackoverflow.com/a/21613044)). On my Mac, `/bin/bash` points to the old 3.x Mac bash version that nobody wants to use.

**Comment by timoreimann** [comment]
Will squash.

File: script/crossbinary (hunk 1 above)
Comment (emilevauge) [inline script/crossbinary]: My question was more : why did you change this in `script/crossbinary` only ?

**Comment by timoreimann** [comment]
Looking at the last build run, I see that it took more than 1000 seconds for the cross-compilation to complete. 😢 

I'll keep the WIP label for now until I'm certain this was just a Travis flake and not a problem with my PR.

Will report back.

File: script/crossbinary (hunk 1 above)
Comment (timoreimann) [inline script/crossbinary]: @emilevauge Ah, good question. The answer is: Because this PR only affects `script/crossbinary` and it failed for me with bash 3.x. :-) (That version doesn't understand `-l` in `declare`.)

My suggestion would be to submit a follow-up PR that changes `/bin/bash` in the other scripts. If you'd prefer the shebang to be changed everywhere atomically, I can revert the part in this script. LMK.

**Comment by timoreimann** [comment]
PR description added.

**Comment by emilevauge** [comment]
@timoreimann is this PR still WIP ?

**Comment by timoreimann** [comment]
@emilevauge unfortunately, I haven't had time yet to verify whether the surprising slow down was due to some Travis-induced flakiness or not, so it's still WIP.

I'll try to give this another whirl some time this week.

**Comment by emilevauge** [comment]
@timoreimann it seems the jobs are now a lot faster right ? :heart_eyes: 
parallel

**Comment by timoreimann** [comment]
@emilevauge not really, unfortunately: I [trimmed down the Travis configuration file](https://github.com/containous/traefik/pull/1231/commits/f387618b6fa6b78d49fa62101326fcd0d59f604c) to have more rapid feedback cycles and limit the CI to just cross compilation.

Looking at the individual runtimes, they don't deviate significantly regardless of whether I run 1, 4, or 6 compilation jobs in parallel (range goes from 1186 to 1207 seconds). It even seems like 1 is doing a bit better than 6. And I can't even run 8 in parallel because Travis kills the job due to lack in progress within a 10 minute time frame.

It seems like running processes concurrently by hand isn't something that can provide any benefit on Travis. I'm afraid I have to close the PR without a positive outcome. :-(

**Comment by emilevauge** [comment]
Ouch :'( Too bad, you did such a great work on this one.

**Comment by timoreimann** [comment]
@emilevauge maybe a different CI provider in the future will give us better results.
